### ISSUE TITLE
Discover help with getting started, esp. compatible file formats

### ISSUE THREAD

**Issue by ebugden** [theia-ide/theia-trace-extension#326]
New users need more help with getting started. Because they don't have a lot invested in the plugin, they need to have smooth onboarding, or they'll give up. 
* Users could struggle with: Finding compatible file formats #324, server configuration, getting certain views to work
* Quick fix: Link to GettingStarted.md or Readme.md from Open Trace dialog (and a link to open the Quick Help panel in other versions)

----
Issue source: @ssmagula's user tests + brainstorming with TC and tool team members

**Comment by ebugden** [comment]
The Linux RT Wiki has some pages it could be fun to point beginners to:
* [RT tools and utilities](https://wiki.linuxfoundation.org/realtime/documentation/howto/tools/start) (ex. [Ftrace](https://wiki.linuxfoundation.org/realtime/documentation/howto/tools/ftrace))
* [Steps for latency debugging](https://wiki.linuxfoundation.org/realtime/documentation/howto/debugging/debug-steps) 

Disclaimer: These were written by pre-information-design Erica so they sound pretentious and are _heavy_ to read

**Comment by ebugden** [comment]
One suggestion is to create a "Getting started with tracing" help page (maybe on the Wiki here) that helps new users get traces if they don't have any. A rough draft for this page is below.

----

# Capture a Quick Kernel Trace

Estimated time: ? (should we we even try to give a time estimate)

* Operating system: Linux
* Tracer: LTTng (Other Linux kernel tracers: Ftrace (point to instructions? ex. [Real time wiki instructions](https://wiki.linuxfoundation.org/realtime/documentation/howto/tools/ftrace)))

### 1. Install LTTng tracer for kernel tracing

The [LTTng installation documentation](https://lttng.org/docs/#doc-installing-lttng) has distribution-specific instructions for how to install.

On Ubuntu, the commands would look something like:

`# apt-get install lttng-tools` (Libraries and command-line interface to control tracing)
`# apt-get install lttng-modules-dkms` (Linux kernel modules to instrument and trace the kernel)

* How to check that it's installed correctly? Is there a lttng --version?

### 2. Trace the Linux kernel using LTTng

The [LTTng kernel tracing quick start](https://lttng.org/docs/#doc-tracing-the-linux-kernel) shows how to enable trace points in the kernel and then collect trace data.

* Which kernel events should we suggest they enable to get a good number of analyses? 
* What operation should we suggest they do while tracing to get a reasonably interesting trace?
* Where should we suggest they put the traces (instead of /tmp)
* Point users to a script to set good defaults?

All together, the commands would look something like:
`# lttng create kernel-session --output=~/quick-kernel-trace`
`# lttng enable-event --kernel sched_switch,sched_process_fork`
`# lttng enable-event --kernel --syscall open,close`
`# lttng start`
[do something exciting like loading a website]
`# lttng destroy`

### 3. Keep your kernel trace in a permanent place
* Make sure they're not in /tmp (disappear when log out)
* Do we want to make ~/traces a thing?

Now you can take a look at your kernel trace with the Trace Viewer! Alternatively, [kernelshark](https://kernelshark.org/) is a well known kernel trace viewer.

**Comment by ebugden** [comment]
Feedback from @arfio 
* **First tracing experience should be something higher level**, userspace (ex. strace, lttng-ust). Maybe introduce them to gantt charts
* **Point users towards a couple basic trace viz labs (for capturing their first traces)**. (ex. [LTTng userspace tracing](https://github.com/tuxology/tracevizlab/tree/master/labs/201-lttng-userspace-tracing), [Kernel trace with Ftrace] (https://github.com/tuxology/tracevizlab/tree/master/labs/004-record-kernel-trace-ftrace))
   * Add a trace viz lab that is a super simple trace and super simple analysis 
   * Completing a couple labs is sufficient for users who are casually exploring the tool, but want to try capturing their own traces
   * The userspace tracing lab is really good for understanding a program, but it shouldn't be a user's first trace 
* **Might be better to point users towards ftrace, perf initially** 
* **Point users to lttng if they're not beginners.** Tracing with lttng is not the best example for tracing beginners
   * Lttng has frequent and difficult to resolve installation problems (ex. modules won't build, modules won't load). These are difficult to resolve even for advanced users.
   * Lttng requires special privileges: Member of tracing group, or sudo for every operation
   * Kernel traces are difficult to decipher for beginners 
   * Point to [Ubuntu installation instructions](https://github.com/tuxology/tracevizlab/tree/master/labs/002-install-lttng-on-ubuntu) in trace viz labs?
   * Point to [record trace helper script](https://github.com/tahini/lttng-utils/blob/master/lttng-record-trace)?
   * To test that lttng is installed correctly: lttng list -k
* If users are exploring the tool because they have a specific problem they want to solve, they are willing to dedicate more time (ex. 1h) to explore, but it's important to have examples (ex. viz labs) that are relevant to their domain / use case. 
* **Look at other analysis tools (ex. more commercial ones) to see how they onboard users.** @arfio Could you suggest a couple for collecting inspiration?

----
Feedback from @ssmagula
* Would userspace tracing be a better first example for beginners?
* Would ftrace or perf traces be a better first trace? Are they compatible with enough views?

**Comment by bhufmann** [comment]
@ebugden the getting started guide depends on the supported trace types in the trace server. The theia-trace-extension front-end is server-agnostic, for example, there might be trace servers that don't support LTTng or fTrace. The domain specific logic and data is in the trace servers. The getting started guide can't be entirely implemented on the front-end side only. I think the trace server needs to provide the detailed information about supported trace types, getting started help etc. That would require a trace server protocol endpoint that the front-end can query and then display to the user. What do you think?

**Comment by ebugden** [comment]
> **bhufmann:** the getting started guide depends on the supported trace types in the trace server. **The theia-trace-extension front-end is server-agnostic**, for example, there might be trace servers that don't support LTTng or fTrace.

True! I'll add this important context to the issue description. Thank you.

> **bhufmann:** I think the **trace server needs to provide the detailed information about supported trace types, getting started help etc.** That would require a trace server protocol endpoint that the front-end can query and then display to the user. What do you think?

For now I'm focusing on defining the content for a reasonably widely applicable help document (tracing on linux with ftrace) that could be tested and iterated on using a wireframe. I'm going to think more seriously about implementation after that. 

More detailed thoughts below.

----

* Having the server send info about supported trace types and getting started help seems like the most well integrated solution, but it also involves changes to the server, tsp and front end which seems like a lot of effort. 
* ~~My first intuition was to initially implement a solution that is less integrated, but easier to implement to allow quick user testing and less complex iteration (ex. Make the front end help link easy to configure (so it can point to a file, external link, or help widget)). But, design iteration could be done with a wireframe instead.~~

Having the trace server provide:
* **Information about supported trace formats and types:** I agree. This makes complete sense based on the server's responsibilities.
   * Can the server easily generate a list based on available analyses? 
* **Getting started help:** It makes sense for the server to provide documentation or to configure help links (based on supported traces). 
   * _More complicated to adjust?:_ Would adjusting the help file that is sent require recompiling the server? After a change to the list of supported trace types, recompilation makes sense, but for changing the help link that is sent recompilation seems like overkill? 
   * _Integrated help (ex. Coach marks):_ The server wouldn't be able to provide more integrated forms of help like coach marks. These would be implemented in the front end, but the server could still be responsible for triggering them (based on supported traces). 

Some less integrated (but easier to implement?) solutions:
* Hardcoded link to Linux-specific help documentation?
* Make the front end help link configurable?
* Have the server specify supported trace types, then configure the help link based on that? (As in, only show ftrace help if ftrace traces are supported) (changes server, tsp, front end)

**Comment by ebugden** [comment]
WIP: 2nd draft for "Getting started with tracing" help page. 

Changes:
* Focus on higher-level (ideally userspace) examples
* Focus on tracers that are included in the kernel: ftrace, perf (don't need to go through a possibly complex installation process).
   * The existing ftrace and perf labs make many mentions of lttng and Trace Compass which would be confusing for new users (even if we say: ignore all these distracting/confusing terms). It could be appropriate to write a slightly edited version of the lab for the quickstart doc. Or rewrite the labs completely but be inspired by them.

Since the UI is moving around so much at this stage, maybe just point towards what tracing is, why it's useful, and how to get some traces. Maybe explain CTF folder structure. They'll have to manage on their own for opening traces and navigation. 

[Record kernel trace with ftrace](https://github.com/tuxology/tracevizlab/tree/master/labs/004-record-kernel-trace-ftrace)
[Record a kernel trace with perf](https://github.com/tuxology/tracevizlab/tree/master/labs/005-record-kernel-trace-perf)
[RT Wiki Ftrace doc](https://github.com/tuxology/tracevizlab/tree/master/labs/005-record-kernel-trace-perf)

**Comment by ebugden** [comment]
This issue came up again during a user-centered planning process.

Need: Help me learn what and how trace  

As a new user, I need some guidance and assistance to get started with trace recording, so that I can go from being a total newb, to a novice more quickly and with less pain.

Provide:
* Tutorial text?
* Bash script(s)?
